tests/unit/htdocs/ConstructorReturnConventionTest.php

```
@@ -0,0 +1,158 @@
+    private const EXCLUDED_PATH_FRAGMENTS = [
+        '/class/libraries/vendor/',
+        '/xoops_lib/vendor/',
+    ];
+
+    /** Every first-party constructor under htdocs/ must yield no value. */
+    public function testNoFirstPartyConstructorReturnsAValue(): void
+    {
+        $violations = [];
+        $scanned    = 0;
+
+        $iterator = new RecursiveIteratorIterator(
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '35', column '25'). (undefined)

(MissingImport)